include/layers/ConvLayer.hpp

```diff
@@ -0,0 +1,62 @@
+#pragma once
+#include <cmath>
+#include <stdexcept>
+#include <vector>
+
+#include "layers/Layer.hpp"
+
+template <typename ValueType>
+class ConvolutionalLayer : public Layer<ValueType> {
+ public:
+  ConvolutionalLayer() = default;
+  std::vector<ValueType> run(
+      const std::vector<ValueType>& input) const override {
+    return input;
+  };
+  void run(const Tensor& input, Tensor& output,
+           const std::vector<ValueType>& kernel, size_t step_);
+};
+template <typename ValueType>
+void ConvolutionalLayer<ValueType>::run(const Tensor& input, Tensor& output,
+                                        const std::vector<ValueType>& kernel_,
+                                        size_t step_) {
+  std::vector<ValueType> matrix = *input.as<ValueType>();
+  size_t input_size = matrix.size();
+  size_t kernel_size = static_cast<int>(std::sqrt(kernel_.size()));
+  int input_width = static_cast<int>(std::sqrt(input_size / 3));
```

Comment:
This doesn't make sense if you can work with shapes, make kernel tensor, and input  have its shape